app/src/main/java/com/hedvig/app/feature/keygear/ui/itemdetail/KeyGearItemDetailViewModel.kt

```diff
@@ -34,36 +36,49 @@ class KeyGearItemDetailViewModelImpl(
 
     override fun loadItem(id: String) {
         viewModelScope.launch {
-            runCatching {
+            val result = runCatching {
                 repository
                     .keyGearItem(id)
+                    .catch { e -> e { e.localizedMessage } }
                     .collect { response ->
                         data.postValue(response.data()?.keyGearItem)
                     }
```

Comment:
Should we use the following form instead?

```suggestion
                    .keyGearItem(id)
                    .onEach { response ->
                        data.postValue(response.data()?.keyGearItem)
                    }
                    .catch { e -> e { e.localizedMessage } }
                    .collect()
```